tests/push/test_http.py

```diff
@@ -673,30 +673,20 @@ def _test_push_unread_count(self):
         )
```

Comment:
there's a comment just above 'This will actually trigger a new notification to be sent out so that even if the user does not receive another message, their unread count goes down'...

It doesn't make much sense to me though because it was **already** at zero right before. 

I would say 'I think it may be worth removing the comment', but this test is trying to test that pushes go out when you read something. As it stands, we're not testing that anymore.
Do you think you can adapt the test so that it will actually get the chance to send a push after a read receipt?

The test currently looks like:

* send message → send count of 0 (because we haven't got a read receipt in the room yet)
* add read receipt → send count of 0 (because we sent a read receipt) — **your changes stop this from happening**, so we don't actually end up testing that read receipts do anything
* send message → send count of 1

I wonder if it should look like this:

* send message → send count of 0 (because we haven't got a read receipt in the room yet)
* add read receipt → **make sure** nothing new has been sent (that's still a valuable piece of test information) — do it with an assert on len(push_attempts)
* send message → send count of 1
* add read receipt (to the second message) → **make sure** we send a count of 0 
* send message → send count of 1

Reply:
Ok, I think I will start working on that issue by the beginning of next week.

Reply:
@reivilibre: I still refactored the test a bit. If you do not like it, the commit before preserves
 the old coding style.